Add helm chart

bramvdklinkenberg opened the pull request:
This PR implements:

- A helm chart to deploy kubenurse
- A helm lint workflow
- Documentation about the helm chart

bramvdklinkenberg commented:
I need some help with this, the ci workflow fails on my own branch. It fails on the GoReleaser step, but I have clue of what is going wrong.

zbindenren commented:
@bramvdklinkenberg the ci workflow is now fixed. Please rebase with master.

Would be great if you update the ci worklfow to deploy the helm chart (similar to the kustomize version) to another namespace and perform the same checks. That would be awesome.

bramvdklinkenberg commented:
> @bramvdklinkenberg the ci workflow is now fixed. Please rebase with master.
> 
> Would be great if you update the ci worklfow to deploy the helm chart (similar to the kustomize version) to another namespace and perform the same checks. That would be awesome.

Hi @zbindenren , I rebased and pushed it to my local branch. 
Do you want me to add the helm deployment in the ci workflow via this PR?
I was also thinking of creating a new PR after this one for creating a helm repo.

zbindenren commented:
> > @bramvdklinkenberg the ci workflow is now fixed. Please rebase with master.
> > Would be great if you update the ci worklfow to deploy the helm chart (similar to the kustomize version) to another namespace and perform the same checks. That would be awesome.
> 
> Hi @zbindenren , I rebased and pushed it to my local branch.
> Do you want me to add the helm deployment in the ci workflow via this PR? And you want the helm deployment done next to the kustomize one?
> I was also thinking of creating a new PR after this one for creating a helm repo.

@bramvdklinkenberg yes that would be great if you include the helm ci integration tests. You should probably create an new namespace in the ci and deploy the helm chart similiar to the kustomize version in that namespace.

Otherwise I have to checkout the branch and test it myself :-)

bramvdklinkenberg commented:
Hi @zbindenren , I added an extra workflow to deploy the helm chart to a kind cluster. Took me a while because I did something wrong with the extraEnv templating in the daemonset. That is fixed now. The workflows succeeded on my own branch

zbindenren reviewed (commented):
Can we discuss the three issues below?

bramvdklinkenberg commented on helm/kubenurse/Chart.yaml:
```diff
@@ -0,0 +1,7 @@
+description: A Helm chart for Kubernetes to deploy kubenurse
+version: 0.1.0
+appVersion: "1.4.0"
+icon: https://www.google.com/imgres?imgurl=https%3A%2F%2Fmedia-exp1.licdn.com%2Fdms%2Fimage%2FC510BAQF_z3JTgqxmdg%2Fcompany-logo_200_200%2F0%2F1519856254399%3Fe%3D2159024400%26v%3Dbeta%26t%3DHi5w0nBq6Y_fVR3acSnTXebGq4tyjJ2UFvuagaEO1Xc&imgrefurl=https%3A%2F%2Fnl.linkedin.com%2Fcompany%2Fpostfinance&tbnid=hMbvcBwLHExq8M&vet=12ahUKEwitmJ_BlNzyAhWWLOwKHdq3BaQQMygAegUIARDEAQ..i&docid=qzrA5TkYH4lMTM&w= ...
```
Adjusted

bramvdklinkenberg commented on README.md:
```diff
@@ -7,6 +7,28 @@ You can get the Docker Image from [Docker Hub](https://hub.docker.com/r/postfina
 contains yamls that deploy the kubenurse to the kube-system namespace.
 
+### Helm deployment
+
```
Will add that to the README with the PR for the helm repo

bramvdklinkenberg commented on .yamllint:
```diff
@@ -0,0 +1,35 @@
+  comments: disable
+  comments-indentation: disable
+  indentation: disable
+  key-duplicates: enable
```
to prevent multiple entries with the same

zbindenren commented:
Thank you.
